rs/index/src/posting_list/file.rs

```diff
@@ -32,23 +31,26 @@ pub struct FileBackedAppendablePostingListStorage {
     resident_posting_lists: Vec<Vec<u64>>,
     resident: bool,
 
-    // Only has value if we spill to disk
+    // Only has value if we spill to disk. We need to separate metadata and posting lists
+    // into 2 different files because num_clusters is not known beforehand.
     base_directory: String,
+    // Backing files for posting list metadata
+    metadata_mmaps: Vec<memmap2::MmapMut>,
+    metadata_current_overall_offset: usize,
+    // Backing files for posting lists themselves
     mmaps: Vec<memmap2::MmapMut>,
-    current_backing_id: i32,
-    current_offset: usize,
-    offset_to_current_posting_list: u64,
+    current_overall_offset: usize,
+    // We currently use overall offset since it would be directly usable by queries to find
+    // the appropriate posting list. We could also use (backing_id, offset_in_file) pair to
+    // track current positions, but we'll need to convert that back to overall offset on
+    // storage's `write`.
+    // TODO(tyb): maybe consider using (backing_id, offset_in_file) pair for consistency with
+    // FileBackedAppendableVectorStorage. Or switch to overall offset for both.
```

Comment:
I think it might be worth it if you create a struct like this:
```
struct MMapsInfo {
    pub mmaps: Vec<...>,
    pub current_overall_offset: usize,
}
```

then, in 
```
struct FileBackedAppendablePostingListStorage {
   metadata_info: MMapsInfo,
   posting_list_info: MMapsInfo,
}
```